django/db/backends/postgresql/base.py

```diff
@@ -427,13 +526,14 @@ def _nodb_cursor(self):
                     connection.vendor == "postgresql"
                     and connection.settings_dict["NAME"] != "postgres"
                 ):
-                    conn = self.__class__(
-                        {
-                            **self.settings_dict,
-                            "NAME": connection.settings_dict["NAME"],
-                        },
-                        alias=self.alias,
-                    )
+                    settings_dict = self.settings_dict.copy()
+                    settings_dict["OPTIONS"] = {
+                        **settings_dict["OPTIONS"],
+                        # Disable pooling for the no-db cursor.
+                        "pool": False,
+                    }
+                    settings_dict["NAME"] = connection.settings_dict["NAME"]
+                    conn = self.__class__(settings_dict, alias=self.alias)
```

Comment:
Thinking out loud here. Would it have any consequences if we set the alias to `NO_DB_ALIAS` here? (Then we'd also would not need to copy the settings and disable pooling since it is disabled by default for the no db alias)

Reply:
This sounds like a cheating to me  :upside_down_face: and can be confusing, e.g. produce misleading database logs.

Reply:
Dunno what the expectation is. One could also expect the no_db_cursor to use the `NO_DB_ALIAS` after all ;) Btw is it necessary to disable the pool for this cursor -- if we are using the existing connection information anyways, we might as well use the pool?

Reply:
> Btw is it necessary to disable the pool for this cursor -- if we are using the existing connection information anyways, we might as well use the pool?

True, looks unnecessary. No crashes (both with and without pooling enabled).